tests/TestCase/Database/Expression/CaseExpressionTest.php

```diff
@@ -53,6 +54,40 @@ public function testSqlOutput()
         $this->assertSame($expected, $caseExpression->sql(new ValueBinder()));
     }
 
+    /**
+     * @return void
+     */
+    public function testWhenThenSqlOutput()
+    {
+        $expr = new QueryExpression();
+        $expr->eq('test', 'true');
+        $expr2 = new QueryExpression();
+        $expr2->eq('test2', 'false');
+
+        $caseExpression = (new CaseExpression())
+            ->when($expr)
+            ->then(new IdentifierExpression('foobar'));
+        $expected = 'CASE WHEN test = :c0 THEN foobar END';
+        $this->assertSame($expected, $caseExpression->sql(new ValueBinder()));
+
+        $caseExpression = (new CaseExpression(new IdentifierExpression('testColumn')))
+            ->when('testValue')
+            ->then('testResult');
+        $expected = 'CASE testColumn WHEN :param0 THEN :param1 END';
+        $this->assertSame($expected, $caseExpression->sql(new ValueBinder()));
+
+        $caseExpression = new CaseExpression();
+        $caseExpression
+            ->when('test1')
+            ->then('test2')
+            ->then('test4')
+            ->when('test3')
```

Comment:
Why are there two then() calls in a row?

Reply:
I test 2 when/then, as long as they're balanced it should work

Reply:
Ok.

Reply:
Was this change merged recently?
Can we please avoid calling it multiple times? I thought we would have a different `MultiCaseExpression` for multiple cases with different syntax. It makes not sense the way it is now

Reply:
This PR introduces this change. I'm not a fan of the when/then calls either (I prefer them in a single call via `add`)

What would be different in the `MultiCaseExpression` than this CaseExpression ? (never saw that in the original issue). Shouldn't both handle a single case with multiple when/then sub-clauses?